### ISSUE TITLE
Incomplete patterns: missing case for constructor `FreshFloat`

### ISSUE THREAD

**Issue by brianhuffman** [GaloisInc/crucible#361]
In `Lang/Crucible/Analysis/Fixpoint.hs` and `Lang/Crucible/Analysis/ForwardDataflow.hs`, case expressions are missing cases for constructor `FreshFloat`:
```
crucible          > /Users/huffman/Work/saw-script/deps/crucible/crucible/src/Lang/Crucible/Analysis/Fixpoint.hs:348:7: warning: [-Wincomplete-patterns]
crucible          >     Pattern match(es) are non-exhaustive
crucible          >     In a case alternative: Patterns not matched: (FreshFloat _ _)
crucible          >     |
crucible          > 348 |       case s of
crucible          >     |       ^^^^^^^^^...
crucible          > [48 of 51] Compiling Lang.Crucible.Simulator.BoundedExec
crucible          > [49 of 51] Compiling Lang.Crucible.Analysis.DFS
crucible          > [50 of 51] Compiling Lang.Crucible.Analysis.ForwardDataflow
crucible          >
crucible          > /Users/huffman/Work/saw-script/deps/crucible/crucible/src/Lang/Crucible/Analysis/ForwardDataflow.hs:175:8: warning: [-Wincomplete-patterns]
crucible          >     Pattern match(es) are non-exhaustive
crucible          >     In an equation for ‘transfer_stmt’:
crucible          >         Patterns not matched: (FreshFloat _ _) _
crucible          >     |
crucible          > 175 |        transfer_stmt (SetReg tp ex) (asgn, c) = (Ctx.extend asgn (kfwd_expr analysis tp ex asgn), c)
crucible          >     |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```

**Comment by robdockins** [comment]
Yeah, these modules need some love.  I don't know offhand who, if anyone, is currently clients for these modules.  @travitch, are you using the dataflow analysis modules for something? We should probably just delete `ForwardDataflow` (which is already deprecated), and figure out how to clean up `Fixpoint`.  There are still a number of features (especially regarding reference cells) that are simply unimplemented.

**Comment by travitch** [comment]
I do have uses of the `Fixpoint` module (in the sfe codebase). That use case doesn't use references at all, so I never implemented anything for them.  I will try to take a look at the float support.

**Comment by travitch** [comment]
It looks like this was probably fixed in #418

**Comment by robdockins** [comment]
Yup, good catch.
